File: Com.Fubei.Api.OpenApi/Biz/OpenApiResponse.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Com.Illuminati.Galileo.Model;
using Newtonsoft.Json.Linq;

namespace Com.Illuminati.Galileo.Biz
{
    /// <summary>
    /// 通用返回结果
    /// </summary>
    /// <typeparam name="T">BaseEntity 子类</typeparam>
    public class OpenApiResponse<T> : BaseEntity
    {
        /// <summary>
        /// 返回结果代码
        /// </summary>
        [JsonProperty("result_code")]
        public int? ResultCode { get; set; }

        /// <summary>
        /// 返回平台错误代码
        /// </summary>
        [JsonProperty("sub_code")]
        public string SubCode { get; set; }

        /// <summary>
        /// 返回结果消息
        /// </summary>
        [JsonProperty("result_message")] 
        public string ResultMessage { get; set; }

        /// <summary>
        /// 返回数据
        /// </summary>
        [JsonProperty("data")] 
        public T Data { get; set; }
    }

    public class OpenApiResponse : OpenApiResponse<JToken>
    {

    }
}

File: Com.Fubei.OpenApi.Vb/Biz/OpenApiResponse.vb
Imports Newtonsoft.Json
Imports Com.Illuminati.Galileo.Model
Imports Newtonsoft.Json.Linq

Namespace Biz
    ' OpenApi返回对象泛型类，1.0和2.0接口
    Public Class OpenApiResponse(Of T)
        Inherits BaseEntity

        <JsonProperty("result_code")>
        Public Property ResultCode As Integer?
        <JsonProperty("sub_code")>
        Public Property SubCode As String
        <JsonProperty("result_message")>
        Public Property ResultMessage As String
        <JsonProperty("data")>
        Public Property Data As T
    End Class

    Public Class OpenApiResponse
        Inherits OpenApiResponse(Of JToken)
    End Class
End Namespace